### DISCUSSION TITLE
A built-in vertex painter

### DISCUSSION THREAD

**Discussion by PrecisionRender** [godotengine/godot-proposals#7819] [3D]
I think Godot should have the ability to paint vertex colors onto 3D models implemented directly into the engine. We do have plugins like [VPainter](https://github.com/tomankirilov/VPainter) that implement this functionality, but they're not without bugs and limitations. (VPainter doesn't officially support 4.0 yet.) Vertex painting is a really useful technique to achieve a higher-quality look in 3D, even with low-poly/retro games. I think it should either be a built-in tool, or more likely, an officially supported addon like Terrain will be.

**Comment by Calinou** [comment]
This requires https://github.com/godotengine/godot-proposals/issues/6486 to be implemented first to be done in a non-destructive way.
